validphys2/src/validphys/correlations.py

```diff
@@ -46,28 +43,29 @@ def _basic_obs_pdf_correlation(pdfarr, obsarr):
 
     return num/den
 
-def _basic_obs_obs_correlation(obsarr1, obsarr2):
+def _basic_obs_obs_correlation(obs1, obs2):
     """Calculate the correlation between two observables. The expected format is
-    obsarr1: (nbins1, nreplicas)
-    obsarr2: (nbins2, nreplicas)
+    Stats instances of:
+
+    obs1: (nreplicas, nbins1)
+    obs2: (nreplicas, nbins2)
 
     The result is (nbins1 , nbins2), a mareix containing the correlation
     coefficients between the two sets.
     """
-    #TODO: Do this at Result level taking into account error type
-    x = (obsarr1.T - np.mean(obsarr1, axis=1)).T
-    y = (obsarr2.T - np.mean(obsarr2, axis=1))
+    x = (obs1.data - obs1.central_value()).T
+    y = (obs2.data - obs2.central_value())
```

Comment:
This doesn't seem universally correct. In particular obviously not for non symmetric hessian sets. Seems to me this would need to be added per-Stats subclass and throw a check error if not implemented. Or more simply (but more fiddly, use `@check_pdf_is_montecarlo`)

Reply:
There is such check, I didn't remove it.

Reply:
Or do you mean it is also wrong for MC PDFs?

Reply:
I should have said that if it is only going to work for monte carlo then it should use arrays rather than stats objects, as it is a bit confusing as it is now.

Reply:
You mean making the input be already "error members" and "central_value"? (for each of the observables). I'm not sure I understand what you suggest to do in this particular case?

Reply:
Originally we had ` def _basic_obs_pdf_correlation(pdfarr, obsarr):` which work with arrays of monte carlo replicas. 

I think it should keep having the same interface (unless it is going to be extended to work for all stats, and even so the mc code would probably call this function...). So I'd suggest reverting the change here and adjusting the caller to pass the array (in the place where we have an explicit check for a mc pdf).

Reply:
I see what you mean now, yes, I can pass `pdf.error_members()` instead of the whole of pdf. It should be fine.